### DISCUSSION TITLE
Auto disconnect after time period to conserve battery life

### DISCUSSION THREAD

**Discussion by calvinli723** [KeyPuncher/WiinUPro#147] [Ideas]
A great quality of life improvement would be to have the Wii U Pro Controller auto-disconnect after, say, 30 minutes of inactivity, in order to preserve battery life. Not sure if there is already a solution for this, but currently what I need to do is turn off bluetooth on my computer, then on reconnect I need to remove the bluetooth device from Windows and then pair it again through WiinUSoft. Even without the auto-disconnect, to have a clean manual disconnect that doesn't require the whole removing bluetooth device process would be a great addition on its own.

I know that this project is only worked on in spare time, but could someone at least show me where I could start looking to see if I can help to implement this feature myself?

**Comment by calvinli723** [comment]
Ah okay, just figured out the "clean manual disconnect" is simply holding down the power button on the controller haha.

**Comment by KeyPuncher** [comment]
Auto disconnect is a good idea. Especially since if it was properly paired a button press should get you back in but that re-connect detection would need to be picked up on to get someone back to their layout.
